ai-server/app/services/music_service.py

```
@@ -0,0 +1,40 @@
+        )
+
+        # (4) Upload generated audio to S3
+        audio_s3_key = await upload_audio(audio, SAMPLE_RATE)
+
+        # (5) Generate presigned URL
+        presigned_url = await generate_presigned_url(audio_s3_key, expires_in=3600)
+
+        return {"s3_key": audio_s3_key, "url": presigned_url}
+
+    except Exception as e:
+        raise AIServerError(f"Music generation pipeline failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)